### PR TITLE
Cover RemoteEmbedder protocol and routing edge cases

### PR DESCRIPTION
**Pull request by iunir**
## What has changed and why?

4/6 of the remote-embedder client stack, now rebased onto `main` since #2358 merged.

The test-only size exception for `RemoteEmbedder`. It grows the smoke tests of #2358 into the
full matrix — capability combinations and registry dispatch, empty, skipped, split and
byte-limited batches, incompatible spaces, dimensions and kept indices, and routes that
disappear or refuse an advertised capability. No production code changes.

## How has it been tested?

`tests/embed/remote/test_embedder.py` passes with 27 tests.

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (tests only)

Suggested reviewer: @michal-lightly; preferably use the same reviewer for the full stack.

## Summary by CodeRabbit

- **Tests**
  - Expanded remote embedding coverage across text, image, and video capabilities.
  - Added validation for capability routing, batching, request-size limits, protocol responses, and unsupported capabilities.
  - Added checks for server loading states and consistency between advertised and available embedding capabilities.

### CODE REVIEW

File: lightly_studio/tests/embed/remote/test_embedder.py [hunk 1]
```diff
@@ -1,65 +1,446 @@
...
+        remote = RemoteEmbedder.connect_with_client(client=server.client())
+        assert isinstance(remote, TextEmbedder)
+        server.capabilities = []
 
-    def test_register_and_resolve(self) -> None:
-        with TestClient(server.create_app(embedder=_TextImageEmbedder())) as client:
+        with pytest.raises(RemoteEmbedderCapabilityError, match="no longer serves text"):
+            remote.embed_text(texts=["a dog"])
+
+        assert server.describe_calls == 2
```
[the remark this answers is not in the document]
Comment (iunir) [reply lightly_studio/tests/embed/remote/test_embedder.py:225]: Good catch, and it went further than the advertised-route tests: no test anywhere asserted a request path, `test_transport.py` included, so a wrong route constant was invisible to the whole `FakeServer` half of the suite.

Fixed in ce724cd0. `FakeServer` now records every path and refuses one the protocol does not name, so this cannot come back per-test, and `test_embed__posts_to_the_route_of_the_capability` reads the route back for text, image and video. Verified by pointing `embed_texts` at the image path — the new test fails.

File: lightly_studio/tests/embed/remote/test_embedder.py [hunk 2]
```diff
@@ -1,63 +1,446 @@
...
+            assert isinstance(remote, TextEmbedder)
+            result = remote.embed_text(texts=["one", "two", "three", "four", "five"])
+
+        assert fake.batches == [["one", "two"], ["three", "four"], ["five"]]
+        assert result.kept_indices == [0, 2, 4]
+        assert result.embeddings.shape == (3, DIMENSION)
+
+    def test_embed_text__other_space_key(self) -> None:
+        body = _embeddings_body(kept_indices=[0], embeddings=[ROW], space_key="acme/model@v2")
+        server = _FakeServer(capabilities=["text"], answer=httpx.Response(200, json=body))
```
[the remark this answers is not in the document]
Comment (iunir) [reply lightly_studio/tests/embed/remote/test_embedder.py]: Done in ce724cd0 — all 8 are `status_code=` now, matching `test_transport.py`.

File: lightly_studio/tests/embed/remote/test_embedder.py [hunk 3]
```diff
@@ -1,63 +1,446 @@
...
+    # Typed as `object`, because the composed class exists only at runtime and mypy reads
+    # `remote` as a plain `RemoteEmbedder`.
+    text_embedder: object = registry.get_text_embedder()
+    image_embedder: object = registry.get_image_bytes_embedder()
+
+    assert text_embedder is remote
+    assert image_embedder is remote
+
+
+class _FakeServer:
```
[the remark this answers is not in the document]
Comment (iunir) [reply lightly_studio/tests/embed/remote/test_embedder.py]: Done in ce724cd0. The fake embedders, `FakeServer` and the body builders moved to `tests/embed/remote/helpers.py`, following `tests/core/mcap/helpers.py`. The test module is 375 lines and the helpers 166.

File: lightly_studio/tests/embed/remote/helpers.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    pass
+
+
+class SkippingTextEmbedder(_FakeEmbedder, TextEmbedder):
+    """Skips the second item of every batch, the way a broken input is skipped."""
+
+    def embed_text(self, texts: list[str]) -> EmbeddingResult:
+        self.batches.append(list(texts))
+        kept_indices = [index for index in range(len(texts)) if index != 1]
+        embeddings = np.array([ROW] * len(kept_indices), dtype=np.float32).reshape(
```
Comment (LeonardoRosaa) [inline lightly_studio/tests/embed/remote/helpers.py:93]: `ROW` has an exact dimension of 2. We don't need to reshape.

File: lightly_studio/tests/embed/remote/helpers.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+
+class FakeTextVideoEmbedder(_FakeEmbedder, TextEmbedder, VideoBytesEmbedder):
+    pass
+
+
+class SkippingTextEmbedder(_FakeEmbedder, TextEmbedder):
+    """Skips the second item of every batch, the way a broken input is skipped."""
+
+    def embed_text(self, texts: list[str]) -> EmbeddingResult:
+        self.batches.append(list(texts))
```
Comment (LeonardoRosaa) [inline lightly_studio/tests/embed/remote/helpers.py:91]: We can append `texts` directly, no? Or do we need a shallow copy? If yes, I'd use `.copy()`.

File: lightly_studio/tests/embed/remote/test_embedder.py [hunk 6]
```diff
@@ -1,53 +1,320 @@
...
-    def embed_text(self, texts: list[str]) -> EmbeddingResult:
-        return _result(count=len(texts))
+    def test_connect__image_bytes_only(self) -> None:
+        with TestClient(server.create_app(embedder=FakeImageEmbedder())) as client:
+            remote = RemoteEmbedder.connect(client=client)
 
-    def embed_image_bytes(self, images: list[bytes]) -> EmbeddingResult:
-        return _result(count=len(images))
+        assert isinstance(remote, ImageBytesEmbedder)
+        assert not isinstance(remote, TextEmbedder)
```
Comment (LeonardoRosaa) [inline lightly_studio/tests/embed/remote/test_embedder.py:56]: Should we assert that the instance is not a `VideoBytesEmbedder`?

Comment (iunir) [reply lightly_studio/tests/embed/remote/test_embedder.py:56]: added assertion

File: lightly_studio/tests/embed/remote/helpers.py (hunk 5 above)
Comment (iunir) [reply lightly_studio/tests/embed/remote/helpers.py]: converted to .copy()

File: lightly_studio/tests/embed/remote/helpers.py (hunk 4 above)
Comment (iunir) [reply lightly_studio/tests/embed/remote/helpers.py]: except for count == 0, where np.array([ROW] * 0) collapses to shape (0,). Replaced both spots with one _stack(count=...) helper built on np.tile, which is (count, 2) at every count, no reshape, and the empty case stays correct.

**Review by LeonardoRosaa — APPROVED** [review]
